UploadServlet.java

```diff
@@ -72,86 +114,96 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
-			
-			try
-			{
-				Class.forName("com.mysql.jdbc.Driver");
+			// if the uploaded file type is PDF, then upload the file to the server
+			if(fileType == "application/pdf") {
 				
-				String u = "root";
-				String password = "Fathihachellam.1";
-				String ul = "jdbc:mysql://localhost:3306/ooad?autoReconnect=true&&useSSL=false";
+				InputStream is = filePart.getInputStream();
+				Files.copy(is, Paths.get(uploadPath + File.separator + fileName), StandardCopyOption.REPLACE_EXISTING);
 				
-				con = DriverManager.getConnection(ul, u, password);
-				
-				String sql = "insert into article(subject, title, highlight, abstracts, author, time, filename, path) values (?,?,?,?,?,?,?,?)";
-				
-				ps = con.prepareStatement(sql);
-				
-				ps.setString(1, subject);
-				ps.setString(2, title);
-				ps.setString(3, highlights);
-				ps.setString(4, abstracts);
-				ps.setString(5, author);
-				ps.setTimestamp(6, time);
-				ps.setString(7, fileName);
-				ps.setString(8, path);
-				
-				int status = ps.executeUpdate();
+				try
+				{
+					Class.forName("com.mysql.jdbc.Driver");
+					
+					String u = "root";
+					String password = "124536";
+					String ul = "jdbc:mysql://localhost:3306/ooad?autoReconnect=true&&useSSL=false";
+					
+					con = DriverManager.getConnection(ul, u, password);
+					
+					String sql = "insert into article(subject, title, highlight, abstracts, author, time, filename, path) values (?,?,?,?,?,?,?,?)";
+					
+					ps = con.prepareStatement(sql);
+					
+					ps.setString(1, subject);
+					ps.setString(2, title);
+					ps.setString(3, highlights);
+					ps.setString(4, abstracts);
+					ps.setString(5, author);
+					ps.setTimestamp(6, time);
+					ps.setString(7, fileName);
+					ps.setString(8, path);
+					
+					int status = ps.executeUpdate();
+					
+					if(status>0)
+					{
+						session.setAttribute("fileName", fileName);
+						String msg = "" +fileName+ " upload successfully";
+						request.setAttribute("msg", msg);
+						request.setAttribute("subject", subject);
+						request.getRequestDispatcher("NewFile.jsp").forward(request, response);
+
+					}
+							
+			}
 				
-				if(status>0)
+				catch(SQLException | ClassNotFoundException e)
 				{
-					session.setAttribute("fileName", fileName);
-					String msg = "" +fileName+ " upload successfully";
-					request.setAttribute("msg", msg);
-					request.setAttribute("subject", subject);
-					request.getRequestDispatcher("NewFile.jsp").forward(request, response);
-//					System.out.println("uploadpath: "+uploadPath);
+					System.out.println(e);
 				}
 				
-			}
-			
-			catch(SQLException | ClassNotFoundException e)
-			{
-				System.out.println(e);
-			}
-			
-			finally
-			{
-				try
+				finally
 				{
-					if(ps != null)
+					try
 					{
-						ps.close();
+						if(ps != null)
+						{
+							ps.close();
+						}
+						
+						if(con!=null)

```

Comment:
@hema-001 

`con!=null` should be `con != null`.  Leave a space before ! and a space after =.

**Hui**

Reply:
In my opinion, this stuff is totally subject to the programmers preferences(coding style).

For me, i prefer spacing a lot for example I would right a loop as follows:
`for(i = 0 ; i < n ; i++)`
In the meantime, @mjhmohamed  might prefer to right it as follows:
`for(i=0;i<n;i++)`

I know it is good for readability to keep some good amount of spaces. But, if the original programmer wrote the whole system not considering these stuff; as a newcomer, as long as I understand what is going on, I wouldn't bother my self with that.

Thanks.

Reply:
> readability

@hema-001 

Readability is one benefit of having adequate spaces.

Consistency is another huge benefit.  It seems @mjhmohamed used "spaces around operator" style too.  For example, see line 177: `if(ps != null)`.

Check the following link for more discussion on this topic.

https://stackoverflow.com/questions/13835769/whats-the-purpose-of-adding-spacings-between-operators

**Hui**